src/java/boa/runtime/BoaRunner.java

```diff
@@ -82,7 +83,9 @@ public Job job(final Path[] ins, final Path out, final boolean robust) throws IO
 		// map output compression
 		configuration.setBoolean("mapred.compress.map.output", true);
 		configuration.set("mapred.map.output.compression.type", "BLOCK");
-		configuration.setClass("mapred.map.output.compression.codec", SnappyCodec.class, CompressionCodec.class);
```

Comment:
Instead of modifying this class, perhaps just overwrite the config value for compression code inside the templated BoaRunner subclass?  That template already has a 'isLocal' template variable so you can easily add the code there without any changes to this file.